doc/OL_preflight_check.py

```diff
@@ -0,0 +1,358 @@
+    """
+    This function is used to check the HttpConfig for OpenLineage events ingestion.
+    extracts and pushes the url to xcom for other tasks to refer.
+    returns the branch based on the backend type. (marquez or other)
+    :param kwargs:
+    """
+    log.info("Config type is HttpConfig")
+    OL_type = kwargs['ti'].xcom_pull(key='OL_type', task_ids='OL_library')
+    if OL_type is None:
+        OL_type = kwargs['ti'].xcom_pull(key='OL_type', task_ids='provider_package')
+    log.debug(f"OL_type: {OL_type}")
+
+    if OL_type == 'OL_library':
+        from openlineage.airflow.adapter import OpenLineageAdapter
+        config = OpenLineageAdapter().client.transport.config
+        url = config.url
+    elif OL_type == 'provider_package':
+        from airflow.providers.openlineage.plugins.listener import get_openlineage_listener
+        config = get_openlineage_listener().adapter.get_openlineage_config()
+        url = config['url']
+    log.info(f"OL_url: {url}")
+    kwargs['ti'].xcom_push(key='url', value=url)
+    if LINEAGE_BACKEND.lower() == "marquez":
+        return 'marquez_check'
+    else:
+        return 'other_server_check'
+
+
+def kafka_config():
+    log.info("Config type is KafkaConfig - TO BE IMPLEMENTED")
+    raise NotImplementedError(
+        "This feature is not implemented yet - Requesting Community to contribute to this feature.")
+
+
+def file_config():
+    log.info("Config type is FileConfig - TO BE IMPLEMENTED")
+    raise NotImplementedError(
+        "This feature is not implemented yet - Requesting Community to contribute to this feature.")
+
+
+def console_config():
+    log.info("Config type is ConsoleConfig - TO BE IMPLEMENTED")
+    raise NotImplementedError(
+        "This feature is not implemented yet - Requesting Community to contribute to this feature.")
+
+
+def marquez_check(**kwargs):
+    """
+    This function is used to check the Marquez setup.
+    It checks if the namespace exists in Marquez.
+    Also checks if Airflow is able to make API calls to the specified lineage backend server.
+    :param kwargs:
+    """
+    log.info("Checking Marquez setup")
+    url = kwargs['ti'].xcom_pull(key='url', task_ids='http_config')
+    list_namespace_url = url + MARQUEZ_PATH_PREFIX + "namespaces"
+    import requests
+    try:
+        response = requests.get(list_namespace_url)
+        response.raise_for_status()
+    except Exception as e:
+        raise Exception(f"Failed to connect to Marquez at {url}", e)
+    log.info("Airflow is able to access the URL")
+    # extract list of all names from list of namespaces
+    namespaces = [namespace['name'] for namespace in response.json()['namespaces']]
+    log.debug(f"All namespaces: {namespaces}")
+
+    OL_type = kwargs['ti'].xcom_pull(key='OL_type', task_ids='OL_library')
+    if OL_type is None:
+        OL_type = kwargs['ti'].xcom_pull(key='OL_type', task_ids='provider_package')
+    log.debug(f"OL_type: {OL_type}")
```

Comment:
In general, there are two possible outputs: we are using provider or not. Maybe let's extract that whole xcom logic somewhere to the function that will answer that question? I see it repeats couple of times.

Reply:
done